src/java/org/apache/cassandra/db/CounterMutation.java

```diff
@@ -175,18 +201,20 @@ private void grabCounterLocks(Keyspace keyspace, List<Lock> locks) throws WriteT
             try
```

Comment:
I think we want to bump the `locksPerUpdate` metric with the number of locks here.
I _think_ the idea for the metric is to know how many locks do we actually need to take, not how many cells do we access.

Reply:
Yes, it's correct and it is explicitly described in the comment.
````
    
/**
     * This metric tracks the number of locks that must be acquired before applying the counter
     * mutation. A mutation normally has one partition only, unless it comes from a batch,
     * where the same partition key is used across different tables.
     * For each partition, we need to acquire one lock for each column on each row.
     * The locks are striped, see {@link CounterMutation#LOCKS} for details.
     */
```